sleuth-camel-core/src/main/java/com.playtika.sleuth.camel/CreatedEventNotifier.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2018 Playtika
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.playtika.sleuth.camel;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.camel.Endpoint;
+import org.apache.camel.Exchange;
+import org.apache.camel.management.event.ExchangeCreatedEvent;
+import org.apache.camel.support.EventNotifierSupport;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.SpanTextMap;
+import org.springframework.cloud.sleuth.Tracer;
+import org.springframework.cloud.sleuth.instrument.messaging.MessagingSpanTextMapExtractor;
+import org.springframework.cloud.sleuth.instrument.messaging.MessagingSpanTextMapInjector;
+import org.springframework.cloud.sleuth.util.SpanNameUtil;
+
+import java.util.EventObject;
+
+import static com.playtika.sleuth.camel.SleuthCamelConstants.FROM_CAMEL;
+
+@Slf4j
+@AllArgsConstructor
+public class CreatedEventNotifier extends EventNotifierSupport {
+
+    private static final String MESSAGE_COMPONENT = "camel";
+
+    private final Tracer tracer;
+    private final MessagingSpanTextMapExtractor spanExtractor;
+    private final MessagingSpanTextMapInjector spanInjector;
+
+    @Override
+    public void notify(EventObject event) {
+        log.trace("Caught ExchangeCreatedEvent, processing...");
+        ExchangeCreatedEvent exchangeCreatedEvent = (ExchangeCreatedEvent) event;
+        Exchange exchange = exchangeCreatedEvent.getExchange();
+        Endpoint endpoint = exchange.getFromEndpoint();
+        CamelMessageTextMap carrier = new CamelMessageTextMap(exchange.getIn());
+
+        Span span = getSpan(carrier, endpoint);
+        span.tag(FROM_CAMEL, "true");
+
+        spanInjector.inject(span, carrier);
+    }
+
+    private Span getSpan(CamelMessageTextMap carrier, Endpoint endpoint) {
```

Comment:
you need to add method parameters into logs

Reply:
didn't want to overload debug logs, added where possible

Reply:
lets assume I'll enable logs on prod and will receive:
```
Built span from message - "hello". Assuming it is remote one. Continuing it...
Built span from message - "hello". Assuming it is remote one. Continuing it...
Built span from message - "hello". Assuming it is remote one. Continuing it...
Built span from message - "hello". Assuming it is remote one. Continuing it...
```

how useful is this??

Reply:
i've already changed logs, you can revisit the code